src/gui/DatabaseWidget.cpp

```diff
@@ -901,12 +938,39 @@ void DatabaseWidget::unlockDatabase(bool accepted)
     }
 }
 
+/**
+ * @author Fonic <https://github.com/fonic>
+ * Add 'copy-on-doubleclick' functionality for certain columns
+ *
+ * TODO:
+ * If pull request #1298 gets merged, double-clicking column 'Attachments'
+ * could open the new details view (see second screenshot)
+ */
 void DatabaseWidget::entryActivationSignalReceived(Entry* entry, EntryModel::ModelColumn column)
 {
-    if (column == EntryModel::Url && !entry->url().isEmpty()) {
-        openUrlForEntry(entry);
+    /* Should never happen */
+    if (!entry) {
+        Q_ASSERT(false);
+        return;
     }
-    else {
+
+    /* Decide what to do based on specified column */
+    switch (column) {
+    case EntryModel::Username:
+        setClipboardTextAndMinimize(entry->resolveMultiplePlaceholders(entry->username()));
+        break;
+    case EntryModel::Password:
+        setClipboardTextAndMinimize(entry->resolveMultiplePlaceholders(entry->password()));
```

Comment:
I think we need to add a little notice to the user that the password was copied, especially if minimize to tray is not enabled. It is not obvious at all that the copy happened. Even adding a little countdown progress bar to the lower right would be awesome to also show the clipboard timeout.

This may be out of scope of this PR, but an idea that came to me while reviewing.

Reply:
Good idea, that's also what KeePass does.

Reply:
agree, canceling this comment